client/src/error.rs

```diff
@@ -53,6 +53,14 @@ pub enum Error {
     /// Invalid BIP32 chain data
     #[error("invalid BIP32 chain data")]
     InvalidBIP32ChainData,
+    /// Invalid bech32 HRP, should match the one from the used network
+    #[error("invalid bech32 hrp for the connected network: {provided_bech32_hrp}, expected: {expected_bech32_hrp}")]
+    InvalidBech32Hrp {
+        /// The bech32 human readable part from the provided address.
+        provided_bech32_hrp: String,
```

Comment:
I'd personally not repeat the bech32_hrp part here and expected but not a big deal, we just usually don't